web/src/app/schedules/ScheduleRuleList.tsx

```diff
@@ -41,27 +42,40 @@ const query = gql`
   }
 `
 
-export default function ScheduleRuleList({ scheduleID }) {
-  const [editTarget, setEditTarget] = useState(null)
-  const [deleteTarget, setDeleteTarget] = useState(null)
-  const [createType, setCreateType] = useState(null)
+interface ScheduleRuleListProps {
+  scheduleID: string
+}
+
+interface TargetValue {
+  id: string
+  type: TargetType
+}
+
+export default function ScheduleRuleList(
+  props: ScheduleRuleListProps,
+): JSX.Element {
+  const { scheduleID } = props
+  const [editTarget, setEditTarget] = useState<TargetValue | null>(null)
+  const [deleteTarget, setDeleteTarget] = useState<TargetValue | null>(null)
+  const [createType, setCreateType] = useState<TargetType | null>(null)
   const isMobile = useIsWidthDown('md')
 
-  const { data, loading, error } = useQuery(query, {
+  const [{ data, fetching, error }] = useQuery({
+    query,
     variables: { id: scheduleID },
-    pollInterval: 0,
   })
+
   const { isLocalZone } = useScheduleTZ(scheduleID)
 
   if (error) {
     return <GenericError error={error.message} />
   }
 
-  if (loading && !data) {
```

Comment:
we can probably remove this entire Spinner block and let it fallback to Suspense